plugins/python/src/main/kotlin/io/paddle/plugin/python/dependencies/GlobalCacheRepository.kt

```diff
@@ -0,0 +1,89 @@
+package io.paddle.plugin.python.dependencies
+
+import com.intellij.util.io.exists
+import io.paddle.plugin.python.Config
+import io.paddle.plugin.python.extensions.Requirements
+import java.nio.file.Files
+import java.nio.file.Path
+
+/**
+ * A service class for managing Paddle's cache.
+ *
+ * @see Config.cacheDir
+ */
+object GlobalCacheRepository {
+    private val cachedPackages: MutableCollection<CachedPackage>
+
+    init {
+        cachedPackages = hashSetOf()
+        Config.cacheDir.toFile().listFiles()?.forEach { packageDir ->
```

Comment:
In that case I would rather use `for` instead of `.forEach`

Reply:
But then I should capture potential NPE from `listFiles()` by myself? It's more convenient to use nested `?.forEach()` here, I guess. But not sure about the performance though..